soc/xtensa/intel_adsp/ace/multiprocessing.c

```diff
@@ -96,6 +96,15 @@ void soc_start_core(int cpu_num)
 		DSPCS.bootctl[cpu_num].battr |= DSPBR_BATTR_LPSCTL_BATTR_SLAVE_CORE;
 	}
 
+	/* Disabling power before core power up. This step is required if core is starting after
+	 * entering power gating and being turn off.
+	 */
+	DSPCS.capctl[cpu_num].ctl &= ~DSPCS_CTL_SPA;
+
+	while (((DSPCS.capctl[cpu_num].ctl & DSPCS_CTL_CPA) == DSPCS_CTL_CPA)) {
```

Comment:
It is worth some comment here about what you are checking. These register names are not easy to udnerstand